contributors/devel/kubemark-guide.md

```diff
@@ -202,9 +234,18 @@ Because there's no easy way of mocking other managers (e.g. VolumeManager), they
 are not supported in Kubemark (e.g. we can't schedule Pods with volumes in them
 yet).
 
-As the time passes more fakes will probably be plugged into HollowNodes, but
-it's crucial to make it as simple as possible to allow running a big number of
-Hollows on a single core.
+We currently plan to extend kubemark along the following directions:
```

Comment:
@xiang90 The master components are isolated in the sense that they all run as separate pods on the master machine unlike before, when they ran as processes under supervisord. However, kubemark as of now doesn't support running these components on different VMs which would require having to create multiple VMs in start-kubemark.sh script and telling each kubelet (if we indeed want to use kubelet) about which components to run and configure networking between them.
Regarding simulating workloads, seems like a pretty interesting idea but could you give a reason why this would be useful (given that all relevant parts of the real cluster are being mimicked by kubemark or at least that is what the current goal is). And IIUC, you mean simulating the requests/responses to/from the apiserver when you say simulating a workload?
(Comments? @wojtek-t @gmarek )

@hongchaodeng Yes, we recently [moved](https://github.com/kubernetes/kubernetes/commit/0a0294c) all the master components to pods. So it should be possible to limit resources for these now.